src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -394,13 +392,21 @@
     <!-- Non-null type is required for this node kind -->
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
 
+    <!-- Various assignable expressions, including some BoundDeconstructionVariables for nested variables -->
     <Field Name="LeftVariables" Type="ImmutableArray&lt;BoundExpression&gt;"/>
 
     <Field Name="Right" Type="BoundExpression"/>
-    <Field Name="DeconstructMemberOpt" Type="MethodSymbol" Null="allow"/>
+
+    <Field Name="DeconstructSteps" Type="ImmutableArray&lt;BoundDeconstructionAssignmentOperator.DeconstructStep&gt;" Null="NotApplicable" SkipInVisitor="true" />
 
     <!-- The assignments have placeholders for the left and right part of the assignment -->
-    <Field Name="Assignments" Type="ImmutableArray&lt;BoundDeconstructionAssignmentOperator.AssignmentInfo&gt;" Null="NotApplicable" SkipInVisitor="true"/>
+    <Field Name="AssignmentSteps" Type="ImmutableArray&lt;BoundDeconstructionAssignmentOperator.AssignmentStep&gt;" Null="NotApplicable" SkipInVisitor="true"/>
+  </Node>
+
+  <Node Name="BoundDeconstructionVariables" Base="BoundExpression">
```

Comment:
Where do we use `checkedVariables` as (array of) `BoundExpression` rather than `BoundNode`? Or is it more than just `checkedVariables` that is the concern?